compiler/rustc_middle/src/ty/context.rs

```diff
@@ -1369,6 +1430,12 @@ impl<'tcx> TyCtxt<'tcx> {
         self.features_query(())
     }
 
+    #[inline(always)]
+    pub fn effects(self) -> bool {
+        self.features().effects
+            || (cfg!(debug_assertions) && std::env::var_os("RUSTC_ENABLE_EFFECTS").is_some())
```

Comment:
`debug_assertions` feels like the wrong check, we should instead check whether this is a nightly build of rustc i think

Reply:
I should probably remove this and add it whenever I do experiments. the env var is not really meant for anything but working on the effect impl